app/models/event.rb

```diff
@@ -1,11 +1,30 @@
 class Event < ApplicationRecord
-  belongs_to :eventable, polymorphic: true
-
   EVENT_NAMES = %w[create update cancel uncancel complete uncomplete redirect lockout].freeze
 
+  belongs_to :eventable, polymorphic: true
+
   validates :eventable, presence: true
   validates :event_name, presence: true, inclusion: { in: EVENT_NAMES }
   validates :client_timestamp, presence: true
+  validates :details, presence: true
 
   scope :default_order, -> { order(client_timestamp: :asc) }
+
+  serialize :details, HashWithIndifferentAccessSerializer
+
+  def supplier_id
+    @supplier_id ||= details.dig(:supplier_id)
```

Comment:
I don't think `dig` is the best choice when only providing one argument - it's a very shallow hole :)

Reply:
👍 good point